aiplatform/src/main/java/aiplatform/EmbeddingModelTuningSample.java

```diff
@@ -0,0 +1,127 @@
+import com.google.protobuf.Value;
+import java.io.IOException;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+// [END aiplatform_sdk_embedding_model_tuning]
+
+public class EmbeddingModelTuningSample {
+  public static void main(String[] args) throws IOException {
+    // [START aiplatform_sdk_embedding_model_tuning]
+    // TODO(developer): Replace these variables before running this sample.
+    String apiEndpoint = "us-central1-aiplatform.googleapis.com:443";
+    String project = "PROJECT";
+    String baseModelVersionId = "BASE_MODEL_VERSION_ID";
+    String taskType = "TASK_TYPE";
+    String pipelineJobDisplayName = "PIPELINE_JOB_DISPLAY_NAME";
+    String outputDir = "OUTPUT_DIR";
+    String queriesPath = "QUERIES";
+    String corpusPath = "CORPUS";
+    String trainLabelPath = "TRAIN_LABEL";
+    String testLabelPath = "TEST_LABEL";
+    int batchSize = 128;
+    int iterations = 1000;
+
+    createEmbeddingModelTuningPipelineJob(
+        apiEndpoint,
+        project,
+        baseModelVersionId,
+        taskType,
+        pipelineJobDisplayName,
+        outputDir,
+        queriesPath,
+        corpusPath,
+        trainLabelPath,
+        testLabelPath,
+        batchSize,
+        iterations);
+    // [END aiplatform_sdk_embedding_model_tuning]
+  }
+
+  // [START aiplatform_sdk_embedding_model_tuning]
+  public static PipelineJob createEmbeddingModelTuningPipelineJob(
+      String apiEndpoint,
+      String project,
+      String baseModelVersionId,
+      String taskType,
+      String pipelineJobDisplayName,
+      String outputDir,
+      String queriesPath,
+      String corpusPath,
+      String trainLabelPath,
+      String testLabelPath,
+      int batchSize,
+      int iterations)
+      throws IOException {
+    Matcher matcher = Pattern.compile("^(?<Location>\\w+-\\w+)").matcher(apiEndpoint);
+    String location = matcher.matches() ? matcher.group("Location") : "us-central1";
+    PipelineServiceSettings settings =
+        PipelineServiceSettings.newBuilder().setEndpoint(apiEndpoint).build();
+    try (PipelineServiceClient client = PipelineServiceClient.create(settings)) {
+      Map<String, Value> parameterValues =
+          Map.of(
+              "project", valueOf(project),
+              "base_model_version_id", valueOf(baseModelVersionId),
+              "task_type", valueOf(taskType),
+              "location", valueOf(location),
+              "queries_path", valueOf(queriesPath),
+              "corpus_path", valueOf(corpusPath),
+              "train_label_path", valueOf(trainLabelPath),
+              "test_label_path", valueOf(testLabelPath),
+              "batch_size", valueOf(batchSize),
+              "iterations", valueOf(iterations));
+      PipelineJob pipelineJob =
+          PipelineJob.newBuilder()
+              .setTemplateUri(
+                  "https://us-kfp.pkg.dev/ml-pipeline/llm-text-embedding/tune-text-embedding-model/v1.1.2")
```

Comment:
Consider pulling this out as a constant next to location at the beginning of the function.

Reply:
Done